src/model/ChillerElectricASHRAE205.cpp

```diff
@@ -0,0 +1,1179 @@
+      return OS_Chiller_Electric_ASHRAE205Fields::OilCoolerOutletNodeName;
+    }
+
+    OptionalModelObject ChillerElectricASHRAE205_Impl::oilCoolerOutletModelObject() const {
+      return connectedObject(ChillerElectricASHRAE205_Impl::oilCoolerOutletPort());
+    }
+
+    boost::optional<Node> ChillerElectricASHRAE205_Impl::oilCoolerOutletNode() const {
+      if (auto mo_ = oilCoolerOutletModelObject()) {
+        return mo_->optionalCast<Node>();
+      }
+      return boost::none;
+    }
+
+    boost::optional<PlantLoop> ChillerElectricASHRAE205_Impl::oilCoolerLoop() const {
+      if (boost::optional<Node> n_ = oilCoolerOutletNode()) {
+        return n_->plantLoop();
+      }
+      return boost::none;
+    }
+
+    bool ChillerElectricASHRAE205_Impl::removeFromOilCoolerLoop() {
+      if (auto plant = oilCoolerLoop()) {
+        return HVACComponent_Impl::removeFromLoop(plant->demandInletNode(), plant->demandOutletNode(), oilCoolerInletPort(), oilCoolerOutletPort());
+      }
+
+      return false;
+    }
+
+    bool ChillerElectricASHRAE205_Impl::addDemandBranchOnOilCoolerLoop(PlantLoop& plantLoop) {
+      Model _model = this->model();
+
+      if (plantLoop.model() != _model) {
+        return false;
+      }
+
+      Splitter splitter = plantLoop.demandSplitter();
+      Mixer mixer = plantLoop.demandMixer();
+
+      if (splitter.outletModelObjects().size() == 1u) {
+        if (boost::optional<ModelObject> mo = splitter.lastOutletModelObject()) {
+          if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+            if ((node->outletModelObject().get() == mixer) && (node->inletModelObject().get() == splitter)) {
+
+              return addToOilCoolerLoopNode(node.get());
+            }
+          }
+        }
+      }
+
+      unsigned nextOutletPort = splitter.nextOutletPort();
+      unsigned nextInletPort = mixer.nextInletPort();
+
+      Node node(_model);
+
+      _model.connect(splitter, nextOutletPort, node, node.inletPort());
+      _model.connect(node, node.outletPort(), mixer, nextInletPort);
+
+      if (addToOilCoolerLoopNode(node)) {
+        return true;
+      }
+
+      _model.disconnect(node, node.outletPort());
+      _model.disconnect(node, node.inletPort());
+      node.remove();
+
+      return false;
+    }
+
+    bool ChillerElectricASHRAE205_Impl::addToOilCoolerLoopNode(Node& node) {
+
+      // One of the example files actually has the Oil Coiler and Auxiliary nodes in series on the same branch...
+      // But this would require a bunch of changes to allow a component to be twice on the same branch, so I'm not allowing it
+      if (node.getImpl<Node_Impl>()->isConnected(getObject<ModelObject>())) {
+        // if ((oilCoolerInletNode() && (node.handle() == oilCoolerInletNode()->handle()))
+        //     || (oilCoolerOutletNode() && (node.handle() == oilCoolerOutletNode()->handle()))) {
+        return false;
+      }
```

Comment:
**IMPORTANT**: modifying everything just to handle this weird use case isn't worth it, so I'm disallowing putting the same component (different ports) on the same branch in series.

Loop::(demand)Components and Loop::findModelObjects don't handle visiting the same component twice...